### PR TITLE
[dotnet] Migrate NUnit assertions to `Assert.That` syntax

### PR DESCRIPTION
**Pull request by RenderMichael**
### **User description**
**Thanks for contributing to Selenium!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) guidelines.
Avoid large PRs, help reviewers by making them as simple and short as possible.

### Description

Migrate NUnit assertions to `Assert.That` syntax

### Motivation and Context

Contributes to #14852 

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

___

### **PR Type**
Tests

___

### **Description**
- Migrated `AlertsTest` and `ChildrenFindingTest` to use modern NUnit syntax.
- Replaced `Assert.AreEqual` with `Assert.That` for better readability and flexibility.
- Improved collection assertions using `Has.Count.EqualTo`.
- Enhanced exception assertions with `Throws.InstanceOf`.
- Utilized `Assert.Multiple` for grouping multiple assertions.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>AlertsTest.cs</strong><dd><code>Migrate AlertsTest to modern NUnit syntax</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

dotnet/test/common/AlertsTest.cs

<li>Migrated from <code>Assert.AreEqual</code> to <code>Assert.That</code> with <code>Is.EqualTo</code>.<br> <li> Improved readability by using <code>Assert.Multiple</code> for multiple assertions.<br> <li> Used <code>Has.Count.EqualTo</code> for collection count assertions.<br> <li> Enhanced exception assertions with <code>Throws.InstanceOf</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/14853/files#diff-f1517edbddef2ed1785c33bd2a05e436e45451e911d2d8b212d646bb4f6e8350">+19/-19</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>ChildrenFindingTest.cs</strong><dd><code>Migrate ChildrenFindingTest to modern NUnit syntax</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

dotnet/test/common/ChildrenFindingTest.cs

<li>Replaced <code>Assert.AreEqual</code> with <code>Assert.That</code> using <code>Is.EqualTo</code>.<br> <li> Utilized <code>Has.Count.EqualTo</code> for collection assertions.<br> <li> Improved exception handling assertions with <code>Throws.InstanceOf</code>.<br> <li> Enhanced readability with <code>Assert.Multiple</code> for grouped assertions.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/14853/files#diff-ea5851f860ebebcecda1c39f2b8f61d6b497996c33f0b4d85a16c9519f3f5148">+65/-43</a>&nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

File: dotnet/test/common/ChildrenFindingTest.cs [hunk 1]
```diff
@@ -41,36 +41,44 @@ public void FindingElementsOnElementByXPathShouldFindTopLevelElements()
...
         {
             driver.Url = nestedPage;
             IWebElement element = driver.FindElement(By.Name("form2"));
+
             ReadOnlyCollection<IWebElement> children = element.FindElements(By.XPath("select/option"));
-            Assert.AreEqual(8, children.Count);
-            Assert.AreEqual("One", children[0].Text);
-            Assert.AreEqual("Two", children[1].Text);
+            Assert.That(children, Has.Exactly(8).Items);
+            Assert.Multiple(() =>
```
Comment (nvborisenko) [inline dotnet/test/common/ChildrenFindingTest.cs]: We all know it is better, but please keep the PR simpler.

**Comment by RenderMichael** [comment]
This PR is ready. Like you said, *big boom*.

**Comment by nvborisenko** [comment]
NP, easy to review. CI doesn't like extra whitespaces in code style.

**Comment by RenderMichael** [comment]
@nvborisenko Sorry for the whitespace! It should work now.

**Comment by RenderMichael** [comment]
Not sure how this slipped by me again. I just ran `scripts\format.sh` locally and it passed, so this time for sure...

**Comment by nvborisenko** [comment]
Turbo-scroll mode has been activated... Thank you, @RenderMichael, for this modernization!

**Pull request merged**